### PR TITLE
Remove button from premium design description in UDP

### PR DESCRIPTION
**Pull request by miksansegundo**
#### Proposed Changes

1. Remove the button element with a direct link to the checkout on premium design descriptions in the Unified Design Picker. We keep the text and remove the button and the direct link when the flag `signup/seller-upgrade-modal` is enabled, or not.

The direct link to the checkout was removed from one of the buttons in https://github.com/Automattic/wp-calypso/pull/67158. Therefore I believe the whole button and the direct link can be removed whether the flag `signup/seller-upgrade-modal` is enabled or not. Note that the flag is disabled in the config for all environments except for wp-calypso.

##### With `signup/seller-upgrade-modal` disabled
|Before|After|
|-------|-----|
|Screen Shot 2565-09-28 at 11 59 37|Screen Shot 2565-09-28 at 13 12 26|

##### With `signup/seller-upgrade-modal` enabled
|Before|After|
|-------|-----|
|Screen Shot 2565-09-28 at 13 22 24|Screen Shot 2565-09-28 at 12 49 49|

2. Thanks to this change, we are fixing the following React warning `<button> can not appear as a descendant of <button>`

image

Screen Shot 2565-09-28 at 11 53 04

#### Testing Instructions

- Go to `/setup/designSetup?siteSlug=[YOUR_SITE]`.
- Find a Premium theme.
- Click on the description `Included in the WordPress.com Premium`.
- Verify that the theme preview is opened with and without this param in the URL `&flags=signup/seller-upgrade-modal`

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to https://github.com/Automattic/wp-calypso/issues/65364

### CODE REVIEW

**Comment by arthur791004** [comment]
cc @Automattic/serenity

**Review by arthur791004 — APPROVED** [review]
LGTM, let's wait for the response from serenity team!

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (taipeicoder) [inline packages/design-picker/src/components/unified-design-picker.tsx:118]: Are we removing the capability of the text being clickable and calling `onCheckout` with this PR?

Comment (arthur791004) [reply packages/design-picker/src/components/unified-design-picker.tsx:118]: According to https://github.com/Automattic/wp-calypso/issues/65364, it would be better to see a more info tip at this point. Therefore, I thought we could remove this behavior as well

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (arthur791004) [inline packages/design-picker/src/components/unified-design-picker.tsx]: I think we need to update the copy. It says “Upgrade to Premium” but I cannot do anything here. When I click on it, the preview will show and it's a little weird. Maybe something like “You need to upgrade to Premium”

But it's not a blocker for this PR

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (taipeicoder) [reply packages/design-picker/src/components/unified-design-picker.tsx:118]: I see, from #67158 it seems intentional to land users to the theme preview instead of the checkout screen. Since it's not tracking click event either, I agree that it's fine to remove the text's role as a button since the entire area is clickable (as it belongs to the theme card).

A small UI difference would be that the text will now be gray instead of link blue. cc: @SaxonF

**Review by taipeicoder — APPROVED** [review]

**Comment by miksansegundo** [comment]
@markbiek as you were the reviewer of https://github.com/Automattic/wp-calypso/pull/67158 created by @andres-blanco, can we ask you to review this PR as well?

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (daledupreez) [reply packages/design-picker/src/components/unified-design-picker.tsx:118]: Apologies for the super-delayed review from our side! 😳 And thanks for taking the time to clean up this React error!

In terms of getting this merged, I have some minor concerns: We're planning on launching this feature later today, and I don't have enough context to know whether we should go to checkout or show the design for the case where we click on this button/link/text. If we do want this to push users through to checkout, then I think we should keep a link with the distinct colour, but I am not sure if we do.
Part of my confusion is that we are doing very closely related work in #69050 to improve the text we show, and that change does _not_ include the click handling. @nelsonec87, can you comment on whether it makes sense to remove this `Button` element completely for this specific case?

Given timing for our relative teams, could you hold off on merging this today? That we we can get some clarification from the Serenity team members in UTC- timezones and clarify what ought to be happening. If you like, we can then merge once we've cleared that up, or confirm if there are changes we'd like for tomorrow. (We can likely make those changes ahead of you all coming in tomorrow if that would be easier for you.)

cc @Automattic/serenity

Comment (miksansegundo) [reply packages/design-picker/src/components/unified-design-picker.tsx:118]: That sounds great! I think we'd happily let you own the PR so you can merge it when you are ready 🙂

Comment (daledupreez) [reply packages/design-picker/src/components/unified-design-picker.tsx:118]: @Automattic/serenity, I am trying to wrap up this PR, and I am not sure whether we want to keep this logic that pushes the user to checkout when they specifically click on the button area. Can someone with more context weigh in on what we should be doing moving forward?

We may also want to clean up the code here around the feature flag now that we've launched the feature.

Comment (markbiek) [reply packages/design-picker/src/components/unified-design-picker.tsx:118]: I'll add another apology for the delayed response. I completely missed the original notification from @miksansegundo 🤦 .

I'm not sure about the correct answer to pushing the user to checkout but have reached out to some other Serenity members for clarity. p1666811036106299-slack-C0Q664T29

File: packages/design-picker/src/components/unified-design-picker.tsx
Comment (markbiek) [reply packages/design-picker/src/components/unified-design-picker.tsx]: Since we've removed the check for the `signup/seller-upgrade-modal`, this copy has also been removed.

**Comment by markbiek** [comment]
@daledupreez I think we're ready to go with this but I'd love if you could give it a quick look before we merge. My thought was to merge and deploy first thing tomorrow if everything looks OK.

**Review by markbiek — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7605628

Thank you @miksansegundo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7605628).

**Pull request merged**
